### PR TITLE
[DXP-645] Streamx run mesh refactoring and tests

### PR DESCRIPTION
**Pull request by jgardo-ds**

### CODE REVIEW

File: src/main/resources/streamx-mesh.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+defaultRegistry: ghcr.io/streamx-dev/streamx-blueprints
+defaultImageTag: latest-jvm # FIXME switch to some specific version of streamx-blueprints
```
[the remark this answers is not in the document]
Comment (jgardo-ds) [reply src/main/resources/streamx-mesh.yml:2]: The problem with latest is that plugin can stop work in moment od deploying non-bug-free version. And invalidation of problematic images may be not so easy for Streamx users, so they will tell just: streamx doesn't work. 

Other problems with latest described [here](https://vsupalov.com/docker-latest-tag/)

I think that bumping versions should be manual.

File: src/main/java/dev/streamx/cli/run/RunConfigResolver.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import jakarta.enterprise.context.ApplicationScoped;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import org.jetbrains.annotations.NotNull;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.ParameterException;
+
+@ApplicationScoped
+class RunConfigResolver {
```
[the remark this answers is not in the document]
Comment (jgardo-ds) [reply src/main/java/dev/streamx/cli/run/RunConfigResolver.java:15]: MeshDefinition sounds good...

File: src/main/java/dev/streamx/cli/run/RunConfigResolver.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+  private static final String CURRENT_DIRECTORY_MESH = "./streamx-mesh.yml";
+  private static final String PREDEFINED_BLUEPRINT_CONFIG = "streamx-mesh.yml";
```
[the remark this answers is not in the document]
Comment (jgardo-ds) [reply src/main/java/dev/streamx/cli/run/RunConfigResolver.java:18]: Well blueprint mesh can be turned on with flag "--blueprints-mesh", so it's not default. It's blueprint.

**Review by kamilchociej — APPROVED** [review]

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,9 @@
+quarkus.package.main-class=dev.streamx.cli.StreamxCommand
+
+quarkus.banner.enabled=false
+quarkus.log.console.level=ERROR
```
[the remark this answers is not in the document]
Comment (jgardo-ds) [reply src/main/resources/application.properties:4]: Will be done within https://teamds.atlassian.net/browse/DXP-689

**Pull request merged**